dev/fig.ts

```diff
@@ -59,6 +59,47 @@ const SETTINGS_GENERATOR: Record<string, Fig.Generator> = {
   "autocomplete.disableForCommands": disableForCommandsGenerator,
 };
 
+interface GithubApiResult {
+  tree: Array<{
+    path: string;
+    type: "blob" | "tree";
+  }>;
+}
+
+const themeGenerator: Fig.Generator = {
+  custom: async (_, executeShellCommand) => {
+    try {
+      const output = await executeShellCommand(
+        'curl -s -H "Accept: application/json" "https://api.github.com/repos/withfig/themes/git/trees/main?recursive=1"'
+      );
+      const builtInThemes = ((JSON.parse(
+        await executeShellCommand(`cat ${SETTINGS_PATH}`)
+      ) as Settings[])?.find((e) => e.settingName === "autocomplete.theme")
+        ?.options ?? []) as string[];
```

Comment:
This might be a bit hard to read and understand. I would prefer something like (not tested):
```suggestion
      const settings: Settings[] = JSON.parse(
        await executeShellCommand(`cat ${SETTINGS_PATH}`)
      ) || [];
      
      const builtInThemes: string[] = settings.find(({ settingName }) => settingName === "autocomplete.theme")?.options ?? [] 
```